frontend/src/lib/components/DateFilter/DateFilter.tsx

```diff
@@ -195,7 +195,8 @@ function DateFilterExperiment({
     dateOptions = dateMappingExperiment,
     isDateFormatted = true,
 }: RawDateFilterProps): JSX.Element {
-    const logicProps = { dateFrom, dateTo, onChange, defaultValue, dateOptions, isDateFormatted }
+    const key = useRef(uuid()).current
```

Comment:
Nice trick! If we'd migrate to React 18, we could use the new `useId` here.

Reply:
Makes me question why we need Kea here but in any case it works 😅

Reply:
Using kea logic with components is a blurry subject, especially in the case of small components that share no state with the outside world. This could have been in pure React, and I wouldn't have minded.

However I do think it's beneficial to keep a strong line between code operating at the application data level (in kea) vs code at the view level (stateless interactive components, react templates around application data). I also know that everything expands in scope, and I've seen too many times how data in that `useState` call eventually wants to interact with data from elsewhere in the app, requiring a rewrite of the component into logic land. Hence I suggest to just save future you the trouble and write a logic directly. 

Since you're considering work that might never happen, it's hard to draw a line. What if we wrote everything with `useState`, and want to go fancy and show a heatmap of events on the calendar, like google flights does for the price? Rewrite to a logic? Parallel system of network calls to load the data in React hooks? 🤷 

Frontend trends come and go, and if we remain consistent in our approach and technologies, we'll build a more maintainable app in the long run.